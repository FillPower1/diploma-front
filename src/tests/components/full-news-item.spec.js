import React from 'react'
import { shallow } from 'enzyme'
import FullNewsItem from '../../components/full-news-item'

describe('Cart Item', () => {
	const props = {
		imgLoad: false
	}

	describe('Full News Item render', () => {

		const nextProps = {
			...props,
			imgLoad: true,
			currentNews: {
				title: "title",
				description: "descr",
				imageSrcL: "",
				dateCreated: new Date().toLocaleDateString()
			}
		}

		const fullNewsItem = shallow(<FullNewsItem {...nextProps} />)

		it('renders properly', () => {
			expect(fullNewsItem).toMatchSnapshot()
		})

		it('render FullNewsItem', () => {
			expect(fullNewsItem.find('Spinner')).toHaveLength(1)
		})

		it('render FullNewsItem title', () => {
			expect(fullNewsItem.find('.full-post-news__title').text()).toEqual(nextProps.currentNews.title)
		})

		it('render FullNewsItem date', () => {
			expect(fullNewsItem.find('.full-post-news__date span').text()).toEqual(nextProps.currentNews.dateCreated)
		})
	})
})