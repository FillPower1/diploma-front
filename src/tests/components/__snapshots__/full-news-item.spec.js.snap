// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Item Full News Item render renders properly 1`] = `ShallowWrapper {}`;
